### ISSUE TITLE
Add r0->r1 calibration to reader

### ISSUE THREAD

**Issue by tibaldo** [cta-observatory/ctapipe_io_nectarcam#22]
Implement prototype of r0->r1 calibration in reader. This will allow to 1) test calibration methods and algorithms, 2) fully process data using ctapipe which requires for some functions to have r1 data.
Will follow implementation for LST.

**Comment by tibaldo** [comment]
I followed as closely as possible the LST implementation. Notes.

- An example calibration file is provided in resources and used for calibration by default and tests. The content was provided by François Brun.
- The calibration file is in the format of a ctapipe MonitoringCameraContainer, which contains a WaveformCalibrationContainer (pedestals and gains) and a FlatFieldContainer (for the moment only relative_mean_gain is filled, the rest will come later).
- By default I disabled the gain selection which is more appropriate at this stage I think. The configuration enables the user to activate gain selection based on maximum waveform value.

Below an example of r0 and r1 waveforms for a pixel (different run from the ones used to generate the calibration coefficients).
![r0r1_example](https://user-images.githubusercontent.com/3046743/182847310-df373c26-48f9-4711-95ec-38670db11ed5.png)

I will make a pull request and let people review the changes before we merge into master. Poke @jlenain @kosack @maxnoe
